mentoring/light_children.py

```diff
@@ -101,12 +101,17 @@ def _is_default(value):
 
     @classmethod
     def init_block_from_node(cls, block, node, attr):
+        from mentoring import MentoringBlock
         block.light_children = []
         for child_id, xml_child in enumerate(node):
             cls.add_node_as_child(block, xml_child, child_id)
 
         for name, value in attr:
-            setattr(block, name, value)
+            if isinstance(block, MentoringBlock):
+                parsed_value = getattr(MentoringBlock, name).from_json(value)
```

Comment:
@antoviaque actually it fixed a couple of other bugs: e.g. submit was still shown, but disabled when display_submit="false". I'll give `enforce_type` a shot, but if I understand right `setattr` just replaced whatever `display_submit` was with a plain `str`.